### PR TITLE
[#4269] doc(flink-connector): add flink connector doc

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

- add flink connector doc
- add flink connector hive doc

### Why are the changes needed?

Fix: #4269

### Does this PR introduce _any_ user-facing change?

- on

### How was this patch tested?

- no need

### CODE REVIEW

File: docs/flink-connector/flink-connector.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+CREATE TABLE hive_students (id INT, name STRING);
+INSERT INTO hive_students VALUES (1, 'Alice'), (2, 'Bob');
+SELECT * FROM hive_students;
+```
+
+## Datatype mapping
+
+Gravitino flink connector support the following datatype mapping between Flink and Gravitino.
+
+| Flink Type                      | Gravitino Type | Since Version |
```
Comment (coolderli) [inline docs/flink-connector/flink-connector.md:69]: the flink type: https://nightlies.apache.org/flink/flink-docs-master/zh/docs/dev/table/types/

**Comment by coolderli** [comment]
@FANNG1 Please help review this MR. Thanks.

**Comment by coolderli** [comment]
@FANNG1 Please help review this MR when you have time. Thanks.

File: docs/flink-connector/flink-connector.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+configuration.setString("table.catalog-store.kind", "gravitino");
+configuration.setString("table.catalog-store.gravitino.gravitino.metalake", "test");
+configuration.setString("table.catalog-store.gravitino.gravitino.uri", "http://localhost:8080");
+EnvironmentSettings.Builder builder = EnvironmentSettings.newInstance().withConfiguration(configuration);
+TableEnvironment tableEnv = TableEnvironment.create(builder.inBatchMode().build());
+```
+
+3. Execute the Flink SQL query. 
+
+There are only one catalog in the metalake `test`.
```
Comment (FANNG1) [inline docs/flink-connector/flink-connector.md:53]: Suppose there is only one hive catalog with the name `hive` in the metalake `test`?

Comment (coolderli) [reply docs/flink-connector/flink-connector.md]: Done

File: docs/flink-connector/flink-connector.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+| `decimal`                       | `decimal`      | 0.6.0         |
+| `double`                        | `double`       | 0.6.0         |
+| `float`                         | `float`        | 0.6.0         |
+| `integer`                       | `integer`      | 0.6.0         |
+| `map`                           | `map`          | 0.6.0         |
+| `null`                          | `null`         | 0.6.0         |
+| `row`                           | `struct`       | 0.6.0         |
+| `smallint`                      | `short`        | 0.6.0         |
+| `time`                          | `time`         | 0.6.0         |
+| `timestamp`                     | `timestamp`    | 0.6.0         |
```
[the remark this answers is not in the document]
Comment (coolderli) [reply docs/flink-connector/flink-connector.md]: There are 5 timestamp types in Flink.
1. timestamp: a timestamp without time zone
2. timestamp without time zone: the same with `timestamp`
3. timestamp with time zone: a timestamp with time zone
4. timestamp with local time zone: Compared to `TIMESTAMP WITH TIME ZONE`, the time zone offset information is not stored physically in every datum. Instead, the type assumes `java.time.Instant` semantics in UTC time zone at the edges of the table ecosystem. 
5. timestamp_ltz: the same with `timestamp with local time zone`

**Comment by coolderli** [comment]
@FANNG1 I have fixed. Please take a look. Thanks.

File: docs/flink-connector/flink-connector.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (coolderli) [reply docs/flink-connector/flink-connector.md]: The `timestamp with time zone` and `timestamp with local time zone` correspond to `timestamp with time zone` in Gravitino. The only difference is the way to obtain the time zone.

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**